eppo/src/main/java/cloud/eppo/android/FlagEvaluator.java

```diff
@@ -0,0 +1,160 @@
+            SubjectAttributes subjectAttributesToEvaluate = new SubjectAttributes(subjectAttributes);
+            if (!subjectAttributesToEvaluate.containsKey("id")) {
+                subjectAttributesToEvaluate.put("id", subjectKey);
+            }
+
+            if (allocation.getRules() != null
+                    && !allocation.getRules().isEmpty()
+                    && RuleEvaluator.findMatchingRule(subjectAttributesToEvaluate, allocation.getRules(), isConfigObfuscated) == null
+            ) {
+                // Rules are defined, but none match
+                continue;
+            }
+
+            // This allocation has matched; find variation
+            for (Split split : allocation.getSplits()) {
+                if (allShardsMatch(split, subjectKey, flag.getTotalShards(), isConfigObfuscated)) {
+                    // Variation and extra logging is determined by the relevant split
+                    variation = flag.getVariations().get(split.getVariationKey());
+                    if (variation == null) {
+                        throw new RuntimeException("Unknown split variation key: "+split.getVariationKey());
+                    }
+                    extraLogging = split.getExtraLogging();
+                    break;
+                }
+            }
+
+            if (variation != null) {
+                // We only evaluate the first relevant allocation
+                allocationKey = allocation.getKey();
+                // doLog is determined by the allocation
+                doLog = allocation.doLog();
+                break;
+            }
+        }
+
+        FlagEvaluationResult evaluationResult = new FlagEvaluationResult();
+        evaluationResult.setFlagKey(flagKey);
+        evaluationResult.setSubjectKey(subjectKey);
+        evaluationResult.setSubjectAttributes(subjectAttributes);
+        evaluationResult.setAllocationKey(allocationKey);
+        evaluationResult.setVariation(variation);
+        evaluationResult.setExtraLogging(extraLogging);
+        evaluationResult.setDoLog(doLog);
+
+        if (isConfigObfuscated) {
+            // Need to unobfuscate the evaluation result
+            evaluationResult.setFlagKey(flagKey);
+            evaluationResult.setAllocationKey(base64Decode(allocationKey));
+            if (variation != null) {
+                Variation decodedVariation = new Variation();
+                decodedVariation.setKey(base64Decode(variation.getKey()));
+                EppoValue decodedValue = EppoValue.nullValue();
+                if (!variation.getValue().isNull()) {
+                    String stringValue = base64Decode(variation.getValue().stringValue());
+                    switch (flag.getVariationType()) {
+                        case BOOLEAN:
+                            decodedValue = EppoValue.valueOf("true".equals(stringValue));
+                            break;
+                        case INTEGER:
```

Comment:
Doesn't Java have an integer type? https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html

Reply:
It does! But `EppoValue` does not distinguish and just loads the JavaScript number, which does not have an integer type.